tests/test_web_urldispatcher.py

```diff
@@ -221,33 +199,33 @@ def teardown():
         path_constructor.return_value = path
 
         # Register global static route:
-        app.router.add_static('/', tmp_dir_path, show_index=True)
+        app.router.add_static('/', str(tmp_path), show_index=True)
         client = await aiohttp_client(app)
 
         # Request the root of the static directory.
-        r = await client.get('/my_dir')
+        r = await client.get('/' + my_dir.name)
         assert r.status == 403
 
 
-async def test_access_symlink_loop(tmp_dir_path, aiohttp_client) -> None:
+async def test_access_symlink_loop(tmp_path, aiohttp_client) -> None:
     """
     Tests the access to a looped symlink, which could not be resolved.
     """
-    my_dir_path = os.path.join(tmp_dir_path, 'my_symlink')
-    os.symlink(my_dir_path, my_dir_path)
+    my_dir_path = tmp_path / 'my_symlink'
+    os.symlink(str(my_dir_path), str(my_dir_path))
```

Comment:
You could use pathlib here as well: https://docs.python.org/3/library/pathlib.html#pathlib.Path.symlink_to

Reply:
I have not been able to get that to work due to pytest-dev/pytest/issues/5017

In short: for Python 3.5, pytest uses `pathlib2` (https://pypi.org/project/pathlib2/) to provide `tmp_path` along with some of the 3.6-style improvements. As a side effect, symlink methods (both `os` and `pathlib`) fail due to isinstance type checking.

Reply:
I mean
```suggestion
    pathlib.Path(str(my_dir_path)).symlink_to(str(my_dir_path))
```

Reply:
Understood! I've fixed all four places in 5928b41